src/util.rs

```diff
@@ -345,6 +345,52 @@ pub unsafe fn get_gc_userdata<T: Any>(state: *mut ffi::lua_State, index: c_int)
     ud
 }
 
+pub unsafe fn init_userdata_metatable_index(
+    state: *mut ffi::lua_State
+) -> Result<()> {
+    match ffi::luaL_dostring(state, cstr!(r#"return function (__index, field_getters, methods)
+        return function (self, key)
+            if field_getters ~= nil and rawget(field_getters, key) ~= nil then
+                return field_getters[key](self)
```

Comment:
I don't think that we can rely on presence of `rawget` method. It can be not available.
Also you do double lookup, first to check the condition then in the block body.